l1-contracts-foundry/script/DeployL1.s.sol

```diff
@@ -0,0 +1,705 @@
+        });
+
+        DiamondInitializeDataNewChain memory initializeData = DiamondInitializeDataNewChain({
+            verifier: IVerifier(addresses.stateTransition.verifier),
+            verifierParams: verifierParams,
+            l2BootloaderBytecodeHash: bytes32(Utils.getBatchBootloaderBytecodeHash()),
+            l2DefaultAccountBytecodeHash: bytes32(Utils.readSystemContractsBytecode("DefaultAccount")),
+            priorityTxMaxGasLimit: config.contracts.priorityTxMaxGasLimit,
+            feeParams: feeParams,
+            blobVersionedHashRetriever: addresses.blobVersionedHashRetriever
+        });
+
+        Diamond.DiamondCutData memory diamondCut = Diamond.DiamondCutData({
+            facetCuts: facetCuts,
+            initAddress: addresses.stateTransition.diamondInit,
+            initCalldata: abi.encode(initializeData)
+        });
+
+        StateTransitionManagerInitializeData memory diamondInitData = StateTransitionManagerInitializeData({
+            governor: config.ownerAddress,
+            validatorTimelock: addresses.validatorTimelock,
+            genesisUpgrade: addresses.stateTransition.genesisUpgrade,
+            genesisBatchHash: config.contracts.genesisRoot,
+            genesisIndexRepeatedStorageChanges: uint64(config.contracts.genesisRollupLeafIndex),
+            genesisBatchCommitment: config.contracts.genesisBatchCommitment,
+            diamondCut: diamondCut,
+            protocolVersion: config.contracts.latestProtocolVersion
+        });
+
+        address contractAddress = deployViaCreate2(
+            abi.encodePacked(
+                type(TransparentUpgradeableProxy).creationCode,
+                abi.encode(
+                    addresses.stateTransition.stateTransitionImplementation,
+                    addresses.transparentProxyAdmin,
+                    abi.encodeCall(StateTransitionManager.initialize, (diamondInitData))
+                )
+            )
+        );
+        console.log("StateTransitionManagerProxy deployed at:", contractAddress);
+        addresses.stateTransition.stateTransitionProxy = contractAddress;
+    }
+
+    function registerStateTransitionManager() internal {
+        Bridgehub bridgehub = Bridgehub(addresses.bridgehub.bridgehubProxy);
+        vm.broadcast();
+        bridgehub.addStateTransitionManager(addresses.stateTransition.stateTransitionProxy);
+        console.log("StateTransitionManager registered");
+    }
+
+    function setStateTransitionManagerInValidatorTimelock() internal {
+        ValidatorTimelock validatorTimelock = ValidatorTimelock(addresses.validatorTimelock);
+        vm.broadcast();
+        validatorTimelock.setStateTransitionManager(
+            IStateTransitionManager(addresses.stateTransition.stateTransitionProxy)
+        );
+        console.log("StateTransitionManager set in ValidatorTimelock");
+    }
+
+    function deployDiamondProxy() internal {
```

Comment:
I believe the approach we took with @Deniallugo for this script was to assume nothing was deployed previously. Therefore we're deploying it here. If necessary an option could be added to the config to pass the address manually. If it's present, then we don't deploy the DiamondProxy, otherwise we deploy it. Do you think this might work?